[Feat] 장소 추천 기능 구현

Jeonkyujin opened the pull request:
## 💡 작업 내용

- [x] 카테고리 기반 장소 조회
- [x] 인기순으로 장소 조회
- [x] 저장된 장소 조회
- [x] 특정 장소 상세 정보 조회
- [x] 현재 위치 기반 장소 조회
- [x] 주소 기반 장소 조회
- [x] 장소 방문 여부 표시
- [x] 친구 기반 장소 조회
- [x] 장소 메모 수정 

## 💡 자세한 설명
- 카테고리 기반 장소 조회 ->사용자가 저장한 장소를 대상으로  요청인자가 CommonPlace 엔티티의 subCategory 필드와 같은 장소 반환
- 인기순으로 장소 조회 ->전체 장소를 대상으로 PlaceStatistics 엔티티의 bookmarkCount 필드값이 큰 순서대로 5개 장소 반환
- 저장된 장소 조회 -> 사용자가 저장한 장소 반환
- 특정 장소 상세 정보 조회 ->  요청인자가 CommonPlace 엔티티의 placeName 필드에 포함되는 장소 반환
- 현재 위치 기반 장소 조회 -> 사용자가 저장한 장소를 대상으로 사용자의 위도, 경도를 요청인자로 받아 거리계산을 통해 값이 작은 순으로 반환(가까운 순으로 반환)
- 주소 기반 장소 조회 -> 사용자가 저장한 장소를 대상으로 요청받은 시/구 단위로 장소를 검색한 뒤, 중복을 제거한 유일한 장소 목록을 반환
- 장소 방문 여부 표시 -> userPlaceId의 요청인자를 통해 UserPlace 엔티티의 visited 필드를 true
- 친구 기반 장소 조회 -> 요청한 사용자와 친구인 사용자의 저장한 장소 반환
- 장소 메모 수정 -> userPlaceId의 요청인자를 통해 memo 필드 수정
## 📗 참고 자료 (선택)
visit과 memo 같이 반환 결과 값이 중요하지 않다고 생각되는 api와 이미 구현되어있었던 장소 생성, 장소 수정, 장소 삭제 api의 사진첨부는 제외시켰습니다.

**[카테고리 기반 장소 조회]**
![카테고리기반](https://github.com/user-attachments/assets/0d8c72ae-db72-45a9-80ae-65e95786fe4e)

**[인기순으로 장소 조회]**
![인기순위기반(bookmarkcount 1위부터 5위까지 순서대로)](https://github.com/user-attachments/assets/e855f151-2840-4603-96f2-55572f31ff08)

**[특정 장소 상세 정보 조회]**
![image](https://github.com/user-attachments/assets/9af72bd5-341a-4ecf-a413-487420ec739e)

**[현재 위치 기반 장소 조회]**
![위치기반](https://github.com/user-attachments/assets/108be47f-29a7-44b3-9968-f5446923a35a)
![위치기반2](https://github.com/user-attachments/assets/1a682d71-7534-4715-b7b2-52e0e78eb31f)

**[주소 기반 장소 조회]**
![주소기반필터(1)](https://github.com/user-attachments/assets/fd55d103-98e2-4387-a11a-c7e6b736d1cd)
![주소기반필터(2)](https://github.com/user-attachments/assets/80a2e189-22f8-4dcb-8bd8-f40ae2cad254)

**[친구 기반 장소 조회]**
![친구 기반(1)](https://github.com/user-attachments/assets/83c25a3e-6f4f-4463-a8c1-d3fd60723114)
![친구기반(2)](https://github.com/user-attachments/assets/2807665a-d890-4043-92a8-b746475fdc33)
**+ 친구 기반 장소 조회 관련 sql**
![sql2](https://github.com/user-attachments/assets/f31b50a9-437e-4cb6-b1e7-fd267de9b75e)
![sql3](https://github.com/user-attachments/assets/197c8141-f016-4615-9436-2c3a6f09d0ea)
![sql1](https://github.com/user-attachments/assets/c75f1f14-7870-42c4-972c-c0c23deca679)

- userId 가 5인 사용자는 6,7 사용자와 친구이고 6,7의 사용자가 저장한 장소를 정상적으로 조회했습니다.

## 📢 리뷰 요구 사항 (선택)
이해가 안되시거나, 구현이 잘못되었다고 생각하시면 편하게 말씀해주세요
## ✅ 셀프 체크리스트
- [x] PR 제목을 형식에 맞게 작성했나요?
- [x] 브랜치 전략에 맞는 브랜치에 PR을 올리고 있나요?
- [x] Assignees, Reviewers, Labels 를 등록했나요?
- [x] 작업 도중 문서 수정이 필요한 경우 잘 수정했나요?
- [x] 불필요한 코드는 제거했나요?

supsup-hae committed:
:sparkles: feat: Jsoup 기반 티스토리 크롤링 전략 구현 (#13)
- Jsoup 라이브러리를 활용한 티스토리 크롤링 구현
- Strategy 패턴을 적용한 크롤링 로직 분리
- 크롤링 대상별 파서 전략 구현

supsup-hae committed:
:sparkles: feat: Spring AI와 OpenAI를 활용한 크롤링 데이터 요약 시스템 구현
- Spring AI 라이브러리를 활용한 OpenAI 연동 구현
- 크롤링 데이터 요약을 위한 프롬프트 엔지니어링
- 요약 결과 처리 및 응답 구조 설계
- AI 요약 서비스 레이어 구현

Jeonkyujin committed:
feat: 카테고리 기반 장소 조회 구현

Jeonkyujin committed:
feat: 카테고리 기반 장소 조회 구현

Jeonkyujin committed:
refactor: CommonPlace 필드 추가 및 UserPlaceResponseDto 수정

Jeonkyujin committed:
refactor: CommonPlace 필드 추가 및 UserPlaceResponseDto 수정

supsup-hae committed:
:sparkles: feat: AI 모델 설정 변경 및 응답 방식 변경(#13)
- AI 모델: gpt 4o -> gpt 3.5 turbo로 변경
- 응답 형식을 List 형식으로 변경하여 반환

Jeonkyujin committed:
feat: 인기 기반 장소 조회 구현

Jeonkyujin committed:
feat: 인기 기반 장소 조회 구현

Jeonkyujin committed:
refactor: UserPlaceResponseDto 매핑을 위한 정적 메소드 추가

Jeonkyujin committed:
refactor: UserPlaceResponseDto 매핑을 위한 정적 메소드 추가

supsup-hae committed:
:sparkles: feat: 네이버 블로그 크롤링 전략 구현 (#13)

supsup-hae committed:
:sparkles: feat: 브런치스토리 크롤링 전략 구현 (#13)

supsup-hae committed:
:sparkles: feat: 기본 페이지 크롤링 전략 구현 (#13)

supsup-hae committed:
:sparkles: feat: 크롤링 전략 비즈니스 코드 로직 분리 및 이미지 추출 기능 구현 (#13)

supsup-hae committed:
:recycle: refactor: AI 요약 응답에 이미지 리스트를 담도록 구조 변경 (#13)

supsup-hae committed:
:recycle: refactor: Response 구조, 네이버 블로그 크롤링 전략 수정 및 모델 변경 (#13)
- gpt 3.5 turbo -> gpt-4o
- 전시정보를 위한 period 추가
- iframe 내부 데이터를 가져오도록 변경

Jeonkyujin committed:
feat: 저장된 장소 조회 기능 구현

Jeonkyujin committed:
feat: 저장된 장소 조회 기능 구현

Jeonkyujin committed:
feat: 특정 장소 상세정보 조회 기능 구현

Jeonkyujin committed:
feat: 특정 장소 상세정보 조회 기능 구현

YoungjaeRo committed:
feat: 이벤트 (Event) Controller 구현

YoungjaeRo committed:
feat: 이벤트 (Event) request, response dto 구현

YoungjaeRo committed:
feat: 이벤트 (Event) service 구현

YoungjaeRo committed:
refactor: 이벤트 (Event) entity에 팩토리 메서드 구현

YoungjaeRo committed:
refactor: 이벤트 (Event) repository 커스텀 쿼리 추가

YoungjaeRo committed:
feat: 이벤트 (Event) Converter 구현

Jeonkyujin committed:
feat: 현재 위치 기반 장소 조회 구현

Jeonkyujin committed:
feat: 현재 위치 기반 장소 조회 구현

Jeonkyujin committed:
feat: 주소 기반 장소 조회 구현

Jeonkyujin committed:
feat: 주소 기반 장소 조회 구현

Jeonkyujin committed:
feat: 장소 방문여부 표시 기능 구현

Jeonkyujin committed:
feat: 장소 방문여부 표시 기능 구현

Jeonkyujin committed:
refactor :트랜잭션 및 영속성 컨텍스트 관리를 위한 Annotation 추가

Jeonkyujin committed:
refactor :트랜잭션 및 영속성 컨텍스트 관리를 위한 Annotation 추가

supsup-hae committed:
:wrench: chore: 사용하지 않는 라이브러리 제거 (#13)

supsup-hae committed:
:recycle: refactor: 크롤링 전략 및 AI 요약 서비스 구조 개선 (#13)
- 크롤링 전략 서비스에 주석 추가
- AI 요약 서비스 로직 문서화
- 공통 크롤링 로직을 CrawlingUtil로 통합
- 로깅 시스템 개선 및 적용

supsup-hae committed:
:recycle: refactor: 패키지명 변경 (#13)

supsup-hae committed:
:recycle: refactor: 클래스 네이밍 변경 및 도메인 구조 변경 (#13)

Jeonkyujin committed:
feat :친구 기반 장소 조회 구현

Jeonkyujin committed:
feat :친구 기반 장소 조회 구현

Jeonkyujin committed:
refactor :API URL 수정 및 로직 주석 추가

Jeonkyujin committed:
refactor :API URL 수정 및 로직 주석 추가

Jeonkyujin committed:
refactor :PlaceResponseDto로 통합

Jeonkyujin committed:
refactor :PlaceResponseDto로 통합

Jeonkyujin committed:
feat :장소 메모 수정 기능 구현

Jeonkyujin committed:
feat :장소 메모 수정 기능 구현

Jeonkyujin committed:
refactor :장소 방문 여부 표시 수정

Jeonkyujin committed:
refactor :장소 방문 여부 표시 수정

YoungjaeRo reviewed (commented):
고생하셨습니다..! 에러 처리랑 converter 패키지도 추가해보면 좋을거 같아요.

Jeonkyujin committed:
refactor: @Tag,@Operation 으로 API 설명 추가

Jeonkyujin committed:
refactor: @Tag,@Operation 으로 API 설명 추가

Jeonkyujin committed:
refactor: placeException 으로 예외처리 추가

Jeonkyujin committed:
refactor: placeException 으로 예외처리 추가

Jeonkyujin committed:
refactor: PathVariable 값 검증 추가

Jeonkyujin committed:
refactor: PathVariable 값 검증 추가

Jeonkyujin committed:
refactor: RequestParam 값 검증 추가

Jeonkyujin committed:
refactor: RequestParam 값 검증 추가

Jeonkyujin committed:
refactor: 장소 삭제 기능 수정

Jeonkyujin committed:
refactor: 장소 삭제 기능 수정

YoungjaeRo committed:
feat: Event 관련 에러처리 초기 구현

YoungjaeRo committed:
refactor: EventController 부연설명 추가

YoungjaeRo committed:
etc: 패키지명 변경

Jeonkyujin committed:
refactor: 장소 생성 기능 수정

Jeonkyujin committed:
refactor: 장소 생성 기능 수정

Jeonkyujin committed:
refactor: placeException 예외처리 수정

Jeonkyujin committed:
refactor: placeException 예외처리 수정

supsup-hae commented on src/main/java/com/adit/backend/domain/place/controller/CommonPlaceController.java:
```diff
@@ -1,68 +1,169 @@
+	@Operation(summary = "장소 생성", description = "카카오 맵 키워드 검색 후 CommonPlace, UserPlace 에 장소를 저장합니다")
+	@PostMapping("/{userId}/create")
+	public ResponseEntity<ApiResponse<PlaceResponseDto>> createPlace(
+		@Valid @RequestBody CommonPlaceRequestDto requestDto,@PathVariable@Min(1) Long userId,@RequestParam String memo) {
```
`@Min(1)`은 어떻게 활용되는걸까요?

Jeonkyujin replied:
userId가 1 미만의 값이라면 예외를 터뜨립니다!

supsup-hae commented on src/main/java/com/adit/backend/domain/place/controller/CommonPlaceController.java:
```diff
@@ -1,68 +1,169 @@
+	public ResponseEntity<ApiResponse<PlaceResponseDto>> createPlace(
+		@Valid @RequestBody CommonPlaceRequestDto requestDto,@PathVariable@Min(1) Long userId,@RequestParam String memo) {
+		if (memo.isBlank()){
+			throw new BusinessException(GlobalErrorCode.NOT_VALID_ERROR);
```
차후에 PlaceException을 정의하면 좋을 거 같아요! 검증등의 비즈니스 로직은 Service단에 명시하는게 좋을듯 합니다.

supsup-hae commented on src/main/java/com/adit/backend/domain/place/exception/placeException.java:
```diff
@@ -0,0 +1,10 @@
+import com.adit.backend.global.error.GlobalErrorCode;
+import com.adit.backend.global.error.exception.BusinessException;
+
+public class placeException extends BusinessException {
```
클래스 네이밍이 잘못 된 것 같습니다!

supsup-hae commented on src/main/java/com/adit/backend/domain/place/repository/UserPlaceRepository.java:
```diff
@@ -0,0 +1,29 @@
+@Repository
+public interface UserPlaceRepository extends JpaRepository<UserPlace, Long> {
+	@Query("SELECT up FROM UserPlace up WHERE up.user.id = :userId AND up.commonPlace.subCategory = :subCategory")
+	List<UserPlace> findByCategory(@Param("subCategory") String subCategory,@Param("userId") Long userId);
```
혹시 필터링하는 카테고리가 여러개일 경우에도 정상적으로 처리가 되는걸까요?

Jeonkyujin replied:
현재는 하나의 카테고리만 처리하도록 구현되어 있습니다!

supsup-hae replied:
제가 알기로는 서비스에서 한번에 여러개의 카테고리를 필터링 할 수 있도록 지원하는 걸로 되어있습니다. 여러개의 필터링에 대해서 적용되도록 한번 검토해주시면 감사하겠습니다!

supsup-hae commented on src/main/java/com/adit/backend/domain/place/controller/CommonPlaceController.java:
```diff
@@ -1,68 +1,169 @@
-		return ResponseEntity.status(HttpStatus.CREATED).body(ApiResponse.success(CommonPlaceResponseDto.from(place)));
-	}
+		CommonPlace place = commonPlaceService.createCommonPlace(requestDto);
+		UserPlace userPlace = commonPlaceService.createUserPlace(place, userId, memo);
```
Service가 처리하는 엔티티를 명확히 구분하는게 좋을 것 같습니다! 그리고 `CommonPlace` 와 `UserPlace`를 정의할때 연관관계 메서드를 엔티티에 정의해서 작성하면 좋을것 같아요!

supsup-hae commented on src/main/java/com/adit/backend/domain/place/controller/CommonPlaceController.java:
```diff
@@ -1,68 +1,169 @@
+		return ResponseEntity.ok(ApiResponse.success("visit sign successfully"));
+	}
+
+	//친구 기반 장소 찾기 API
```
친구 엔티티 관련 해서는 해당 엔티티 개발때 통합하려고했는데 미리 작성해주셔서 감사합니다!

supsup-hae commented on src/main/java/com/adit/backend/domain/place/dto/response/PlaceResponseDto.java:
```diff
@@ -0,0 +1,108 @@
+			userPlace.getCommonPlace().getRoadAddressName(),
+			userPlace.getCommonPlace().getAddressName(),
+			null,
+			userPlace.getCommonPlace().getImages().stream().map(Image::getUrl).collect(Collectors.toList()),
```
java17 버전부터 Stream한 결과를 List로 바꿀때 `toList()`로 간단히 바꿀 수 있습니다!

supsup-hae commented on src/main/java/com/adit/backend/domain/place/dto/response/PlaceResponseDto.java:
```diff
@@ -0,0 +1,108 @@
+import jakarta.annotation.Nullable;
+import software.amazon.ion.Decimal;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
```
응답값이 정해지지 않아서 우선 엔티티 전체를 반환하신것처럼 보입니다. 해당 클래스는 이후 프론트 측과 이야기 후에 전체적으로 수정하면 될것 같아요!

supsup-hae commented on src/main/java/com/adit/backend/domain/place/dto/response/PlaceResponseDto.java:
```diff
@@ -0,0 +1,108 @@
+			userPlace.getMemo(),
+			userPlace.getVisited(),
+			userPlace.getCommonPlace().getPlaceName(),
+			null,
```
값이 null 인부분인 삼항 연산자를 통해서 값 존재 유무에 따라서 가져오거나 검증 메서드를 엔티티에 추가해서 명시하면 될것 같아요!

Jeonkyujin replied:
값은 무조건 존재하는데, 디자인 페이지 상황을 참고하여 필요없다고 느낀 필드를 null로 지정한 상태입니다!

supsup-hae commented on src/main/java/com/adit/backend/domain/place/dto/response/PlaceResponseDto.java:
```diff
@@ -0,0 +1,108 @@
+			.toList();
+
+	}
+
```
빈 공백은 PR 진행전 검토 부탁드립니다! 코드 서식 정리 단축키를 사용하시면 공통적으로 코드가 깔끔히 정리될거에요!

supsup-hae commented on src/main/java/com/adit/backend/domain/place/service/CommonPlaceService.java:
```diff
@@ -20,9 +36,12 @@
 
 	// 새로운 장소 생성
-	public CommonPlace createPlace(CommonPlaceRequestDto requestDto) {
+	public CommonPlace createCommonPlace(CommonPlaceRequestDto requestDto) {
```
Converter에 대해서는 영재님께서 답해주신것 같아서 해당부분은 참고해서 변환해주시면 될것 같아요! 그리고 Service 단에서 오가는 데이터는 Dto 형식으로 정의 부탁드립니다.

supsup-hae commented on src/main/java/com/adit/backend/domain/place/service/CommonPlaceService.java:
```diff
@@ -59,10 +93,140 @@ public CommonPlace updatePlace(Long placeId, CommonPlaceRequestDto requestDto) {
+	}
+
+	//카테고리 기반으로 장소 찾기
+	@Transactional(readOnly = true)
```
지금 프로젝트에서 CQS 패턴을 적용중입니다! 해당 개념 참고하셔서 리팩토링할때 반영해주시면 될 것 같아요!
[CQS 패턴](https://velog.io/@yena1025/CQS-Command-Query-Separation)

supsup-hae commented on src/main/java/com/adit/backend/domain/place/service/CommonPlaceService.java:
```diff
@@ -59,10 +93,140 @@ public CommonPlace updatePlace(Long placeId, CommonPlaceRequestDto requestDto) {
+	@Transactional(readOnly = true)
+	public List<PlaceResponseDto> getPlaceByPopular() {
+		//PlaceStatistics 엔티티에서 1위부터 5위까지의 commonplaceId를 가져옴
+		Pageable pageable = PageRequest.of(0,5);
```
해당 부분은 상수처리 가능할것 같네요! 이후 프로젝트에서 요구하는 인기기반 장소의 수가 바뀔 수 있으니 유지보수를 위해 상수로 적용하면 좋을 것 같습니다.

supsup-hae commented on src/main/java/com/adit/backend/domain/place/service/CommonPlaceService.java:
```diff
@@ -59,10 +93,140 @@ public CommonPlace updatePlace(Long placeId, CommonPlaceRequestDto requestDto) {
+		//PlaceStatistics 엔티티에서 1위부터 5위까지의 commonplaceId를 가져옴
+		Pageable pageable = PageRequest.of(0,5);
+		List<Long> commonPlacesId = commonPlaceRepository.findByPopular(pageable);
+		if (commonPlacesId.isEmpty()){
```
현재 `Place`를 검증하는 부분이 아래 `commonPlaces` 를 매핑하는 과정이랑 중복 된 것 같아요!

supsup-hae commented on src/main/java/com/adit/backend/domain/place/service/CommonPlaceService.java:
```diff
@@ -59,10 +93,140 @@ public CommonPlace updatePlace(Long placeId, CommonPlaceRequestDto requestDto) {
+	}
+
+	//현재 위치 기반 장소 찾기
+	@Transactional(readOnly = true)
```
해당 로직은 회의때 이야기를 나누면 좋을것 같습니다.

supsup-hae commented on src/main/java/com/adit/backend/domain/place/service/CommonPlaceService.java:
```diff
@@ -59,10 +93,140 @@ public CommonPlace updatePlace(Long placeId, CommonPlaceRequestDto requestDto) {
+	//주소 기반 장소 찾기
+	@Transactional(readOnly = true)
+	public List<PlaceResponseDto> getPlaceByAddress(List<String> address, Long userId) {
+		//중복 제거를 위한 Set
```
Place의 중복제거는 어떤 상황때 적용되는 걸까요!

Jeonkyujin replied:
예를들어서, 서울특별시 마포구라는 주소를 가진 장소가 있다면 이 장소는 서울특별시 필터링에도 포함되고 마포구 필터링에도 포함이 됩니다. 이런 경우 중복이 발생하여 2개 이상의 장소가 반환될 수 있으므로, 중복된 항목은 제거됩니다!

supsup-hae commented on src/main/java/com/adit/backend/global/error/GlobalErrorCode.java:
```diff
@@ -112,7 +112,14 @@ public enum GlobalErrorCode implements ErrorCode {
 
-	FIELD_NOT_FOUND(NOT_FOUND, "4002", "해당 필드를 찾지 못했습니다");
+	FIELD_NOT_FOUND(NOT_FOUND, "4002", "해당 필드를 찾지 못했습니다"),
+
```
현재 엔티티 별 코드를 별도로 구분하지 않았다 보니 중복되는 경우가 발생하는 것 같습니다. 회의때 이야기 나눠봐요!

supsup-hae reviewed (changes requested):
수고하셨습니다! 리뷰 검토 부탁드릴게요!

Jeonkyujin committed:
refactor: placeException 예외처리 재수정

Jeonkyujin committed:
refactor: placeException 예외처리 재수정

Jeonkyujin committed:
refactor: 카테고리 기반 장소 조회 로직 수정

Jeonkyujin committed:
refactor: 카테고리 기반 장소 조회 로직 수정

Jeonkyujin committed:
refactor: 연관관계 메소드 활용하여 장소 생성 로직 수정

Jeonkyujin committed:
refactor: 연관관계 메소드 활용하여 장소 생성 로직 수정

Jeonkyujin committed:
refactor: CQS 패턴 적용 및 컨버터 구현으로 코드 구조 개선

Jeonkyujin committed:
refactor: CQS 패턴 적용 및 컨버터 구현으로 코드 구조 개선

Jeonkyujin committed:
refactor: 상수화 및 코드정리

Jeonkyujin committed:
refactor: 상수화 및 코드정리

YoungjaeRo committed:
refactor: entity에 update 로직 추가

YoungjaeRo committed:
refactor: EventConverter 에서 Entity의 updateEntity를 호출할 수 있게 수정ㄷㄷㄷㄷㄷㄷㄷㄷㄷㄷㄷㄷㄷㄷㄷㄷㄷㄷㄷㄷㄷㄷㄷ

YoungjaeRo committed:
refactor : event 엔터티 업데이트에 관한 로직 플로우 수정

supsup-hae committed:
:twisted_rightwards_arrows: Merge pull request #32 from IT-Cotato/feat/28

[Feat] 이벤트(전시/공연) API 구현 완료

supsup-hae committed:
🔨 Chore: 아티팩트 액션 버전 업그레이드 v3 -> v4  

- actions/upload-artifact 및 download-artifact v3에서 v4로 업그레이드 
- JAR 파일 업로드 시 compression-level 옵션 추가하여 성능 최적화

Jeonkyujin committed:
Merge pull request #34 from IT-Cotato/feature/33

[fix] CI/CD 중단 오류 해결

YoungjaeRo committed:
bug : Event 관련 에러 GlobalExceptionHandler 에서일시적 등록 해제

supsup-hae committed:
Merge pull request #36 from IT-Cotato/feat/28

Fix : Event 관련 에러 GlobalExceptionHandler에서 일시적 등록 해제

supsup-hae committed:
:sparkles: feat: LoggingAdvisor 구현 및 적용 (#13)
- 토큰 사용량 추적을 위한 Advisor 추가
- AI 설정에 InMemoryChatMemory 값 추가

supsup-hae committed:
:sparkles: feat: Swagger 문서 개선 (#13)
- AI Summary API Swagger 문서 설명 추가

supsup-hae committed:
:recycle: refactor: 메서드 네이밍 변경 (#13)
- `analyzeCulturalInfo` -> `summaryContent`

supsup-hae committed:
:memo: docs: LoggingAdvisor 주석 추가 (#13)

supsup-hae committed:
:fire: remove: 미사용 클래스 제거 (#13)

supsup-hae committed:
:recycle: refactor: 상수 접근제어자 변경 (#13)

supsup-hae committed:
:fire: remove: 미사용 코드 제거 (#13)

supsup-hae committed:
Merge branch 'develop' into feature/13

supsup-hae committed:
:twisted_rightwards_arrows: Merge pull request #37 from IT-Cotato/feature/13

[Feat] OpenAI API를 이용한 링크 요약 시스템 구현 완료

supsup-hae committed:
:recycle: refactor: 구문 오류 제거

supsup-hae committed:
Merge remote-tracking branch 'IT-Cotato/feature/14' into feature/14

# Conflicts:
#	src/main/java/com/adit/backend/global/error/GlobalErrorCode.java
#	src/main/java/com/adit/backend/global/error/GlobalExceptionHandler.java

supsup-hae committed:
:fire: remove: rebase 중복 코드 제거 (#14)

supsup-hae committed:
:fire: remove: 미사용 코드 제거 (#14)
